### PR TITLE
[ios_static_routes] - fix vrf for ipv6 route command

### LINKED ISSUE #660: ios_static_routes does not configure IPv6 routes in VRF
##### SUMMARY
Static IPv6 routes are created in the default routing context instead of the specified VRF.

Playbook - IPv6 routes shall be created in VRF blue:

```
---

- name: Configure static routes
  hosts: CSR1Kv-0
  connection: local
  gather_facts: false

  tasks:
  - name: Create static routes
    cisco.ios.ios_static_routes:
      config:
      - vrf: blue
        address_families:
        - afi: ipv6
          routes:
          - dest: FD3D:3D:A:16::/64
            next_hops:
            - forward_router_address: FD3D:3D:A:14::2
          - dest: FD3D:3D:A:17::/64
            next_hops:
            - forward_router_address: FD3D:3D:A:14::2
```

Result - However routes are incorrectly created in the default routing context:
```
sr1000v-1#show running-config | include ip route|ipv6 route
<...>>
ipv6 route FD3D:3D:A:16::/64 FD3D:3D:A:14::2
ipv6 route FD3D:3D:A:17::/64 FD3D:3D:A:14::2
```

In case of IPv4, static routes are correctly configured in the specified VRF, i.e. the issue is specific to IPv6.

Cross-check: When using the ios_config module instead of the ios_static_routes modules, the routes are configured correctly:

Playbook:
```
---

- name: Configure static routes
  hosts: CSR1Kv-0
  connection: local
  gather_facts: false

  tasks:
  - name: Create static routes
    cisco.ios.ios_config:
      lines:
      - ipv6 route vrf blue FD3D:3D:A:16::/64 FD3D:3D:A:14::2
      - ipv6 route vrf blue FD3D:3D:A:17::/64 FD3D:3D:A:14::2
```
Result:
```
csr1000v-1#show running-config | include ip route|ipv6 route
<...>
ipv6 route vrf blue FD3D:3D:A:16::/64 FD3D:3D:A:14::2
ipv6 route vrf blue FD3D:3D:A:17::/64 FD3D:3D:A:14::2
```

Therefore I think this is a bug in the ios_static_routes module.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
cisco.ios.ios_static_routes

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.1]                                                                                                                                
  config file = /home/marehler/Documents/Ansible/ansible.cfg                                                                                         
  configured module search path = ['/home/marehler/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']                                  
  ansible python module location = /usr/local/lib/python3.10/dist-packages/ansible                                                                   
  ansible collection location = /home/marehler/.ansible/collections:/usr/share/ansible/collections                                                   
  executable location = /usr/local/bin/ansible                                                                                                       
  python version = 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]                                                                                 
  jinja version = 3.1.2                                                                                                                              
  libyaml = True
```

##### COLLECTION VERSION

```paste below
# /usr/local/lib/python3.10/dist-packages/ansible_collections                                                                                        
Collection Version                                                                                                                                   
---------- -------                                                                                                                                   
cisco.ios  3.2.0 
```

##### CONFIGURATION

```paste below
DEFAULT_HOST_LIST(/home/marehler/Documents/Ansible/ansible.cfg) = ['/home/marehler/Documents/Ansible/inventory/hosts.ini']                           
HOST_KEY_CHECKING(/home/marehler/Documents/Ansible/ansible.cfg) = False                                                                              
PERSISTENT_COMMAND_TIMEOUT(/home/marehler/Documents/Ansible/ansible.cfg) = 30                                                                        
PERSISTENT_CONNECT_TIMEOUT(/home/marehler/Documents/Ansible/ansible.cfg) = 60 
```

##### OS / ENVIRONMENT

Cisco IOS XE Software, Version 17.03.04a                                                                                                             
Cisco IOS Software [Amsterdam], Virtual XE Software (X86_64_LINUX_IOSD-UNIVERSALK9-M), Version 17.3.4a, RELEASE SOFTWARE (fc3)                         
                                                                                                                                  
Cisco IOS Software, IOSv Software (VIOS-ADVENTERPRISEK9-M), Version 15.9(3)M4, RELEASE SOFTWARE (fc3) 

##### STEPS TO REPRODUCE

```yaml
---

- name: Configure static routes
  hosts: CSR1Kv-0
  connection: local
  gather_facts: false

  tasks:
  - name: Create static routes
    cisco.ios.ios_static_routes:
      config:
      - vrf: blue
        address_families:
        - afi: ipv6
          routes:
          - dest: FD3D:3D:A:16::/64
            next_hops:
            - forward_router_address: FD3D:3D:A:14::2
          - dest: FD3D:3D:A:17::/64
            next_hops:
            - forward_router_address: FD3D:3D:A:14::2
```

##### EXPECTED RESULTS

```
ipv6 route vrf blue FD3D:3D:A:16::/64 FD3D:3D:A:14::2
ipv6 route vrf blue FD3D:3D:A:17::/64 FD3D:3D:A:14::2
```

##### ACTUAL RESULTS

```paste below
ipv6 route FD3D:3D:A:16::/64 FD3D:3D:A:14::2
ipv6 route FD3D:3D:A:17::/64 FD3D:3D:A:14::2
```

### PR DESCRIPTION
**Pull request by bentole**
##### SUMMARY
Vrf-support for IPv6 routes.

fixes #660 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins.module_utils.network.ios.config.static_routes.static_routes

##### ADDITIONAL INFORMATION
The fix is verifed on IOS-XE version 17.3.

I havent verified 100% if there are more places that needs changing, but at least the proposed changes makes the ipv6 route command work properly for vrfs.

### CODE REVIEW

File: changelogs/fragments/664-ipv6-vrf-fix.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - fixes vrf support for ipv6 static routes
```
Comment (Qalthos) [inline changelogs/fragments/664-ipv6-vrf-fix.yaml:2]: Please update this changelog fragment to follow our guidelines (https://docs.ansible.com/ansible/latest/community/development_process.html#changelog-fragment-entry-format).

**Review by Qalthos — CHANGES_REQUESTED** [review]
Can you update the unit and integration tests to cover this?

File: changelogs/fragments/664-ipv6-vrf-fix.yaml (hunk 1 above)
Comment (bentole) [reply changelogs/fragments/664-ipv6-vrf-fix.yaml:2]: Hi Kate, I'll update the fragment and tests. Apologies for not doing so (properly) already.. :)

File: changelogs/fragments/664-ipv6-vrf-fix.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - cisco.ios.ios_static_routes - fix vrf for ipv6 static routes.
```
Comment (Qalthos) [inline changelogs/fragments/664-ipv6-vrf-fix.yaml:2]:
```suggestion
  - ios_static_routes - fix vrf for ipv6 static routes. (https://github.com/ansible-collections/cisco.ios/issues/660)
```

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Review by KB-perByte — APPROVED** [review]

**Pull request merged**